java/vector/src/main/java/org/apache/arrow/vector/BaseNullableFixedWidthVector.java

```diff
@@ -0,0 +1,500 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector;
+
+import com.google.common.base.Preconditions;
+
+import io.netty.buffer.ArrowBuf;
+import java.util.List;
+
+import org.apache.arrow.memory.OutOfMemoryException;
+import org.apache.arrow.memory.BaseAllocator;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.util.OversizedAllocationException;
+
+public abstract class BaseNullableFixedWidthVector implements FixedWidthVector, FieldVector {
+   private static final org.slf4j.Logger logger =
+           org.slf4j.LoggerFactory.getLogger(NullableIntVector.class);
+   public static final String MAX_ALLOCATION_SIZE_PROPERTY = "arrow.vector.max_allocation_bytes";
+   public static final int MAX_ALLOCATION_SIZE = Integer.getInteger(MAX_ALLOCATION_SIZE_PROPERTY,
+           Integer.MAX_VALUE);
+   public static final int INITIAL_VALUE_ALLOCATION = 4096;
+
+   private final String name;
+   private final BufferAllocator allocator;
+   private final int typeWidth;
+   private final Field field;
+
+   private int valueAllocationSizeInBytes;
+   private int validityAllocationSizeInBytes;
+
+   protected ArrowBuf validityBuffer;
+   protected ArrowBuf valueBuffer;
+   protected int allocationMonitor;
+   protected int valueWriteIndex;
+   protected int valueCount;
+
+   public BaseNullableFixedWidthVector (final String name, final BufferAllocator allocator,
+                                        FieldType fieldType, final int typeWidth) {
```

Comment:
Is it possible to get `typewidth` from `fieldtype`?  Or maybe add a mappng in `ArrowType` that you could query here?

Reply:
Nvm, I guess we need the TYPE_WIDTH in the concrete class for get/set methods anyway